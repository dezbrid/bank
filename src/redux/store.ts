import {
  configureStore,
  ThunkAction,
  Action,
  StoreEnhancer,
} from '@reduxjs/toolkit';
import AsyncStorage from '@react-native-async-storage/async-storage';
//import storage from 'redux-persist/lib/storage';
import {combineReducers} from 'redux';
import {persistReducer} from 'redux-persist';
import ReactotronConfig from '@config/ReactotronConfig';
import bankSlice from '@redux/bankSlice';
import thunk from 'redux-thunk';

const enhancers: StoreEnhancer[] = [];
if (ReactotronConfig.createEnhancer) {
  enhancers.push(ReactotronConfig.createEnhancer());
}

const reducers = combineReducers({
  bank: bankSlice,
});

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: [thunk],
  enhancers,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
