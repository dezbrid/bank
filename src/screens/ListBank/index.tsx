import React, {useEffect} from 'react';
import {FlatList, SafeAreaView, ListRenderItem, View} from 'react-native';
import {useAppDispatch, useAppSelector} from '@redux/hooks';
import {bankAsync, listBank, orignialListBank} from '@redux/bankSlice';
import {BankObject} from '@interfaces/bank';
import {ListKeyExtractor} from '@interfaces/generic';

import Bank from './components/Bank';
import BarSearch from './components/BarSearch';

import styles from './styles';

function ListBank() {
  const dispatch = useAppDispatch();
  const banks = useAppSelector(listBank);
  const originalBank = useAppSelector(orignialListBank);
  useEffect(() => {
    if (orignialListBank.length === 0) {
      dispatch(bankAsync());
    }
  }, [dispatch, originalBank]);
  const renderItem: ListRenderItem<BankObject> = ({item}) => <Bank {...item} />;
  const keyExtractor: ListKeyExtractor<BankObject> = (_, i) => i.toString();
  const separator = () => <View style={styles.separator} />;
  return (
    <SafeAreaView style={styles.container}>
      <BarSearch />
      <FlatList<BankObject>
        data={banks}
        renderItem={renderItem}
        keyExtractor={keyExtractor}
        contentContainerStyle={styles.contentContainer}
        ItemSeparatorComponent={separator}
      />
    </SafeAreaView>
  );
}

export default ListBank;
