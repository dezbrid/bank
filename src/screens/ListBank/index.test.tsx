import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';

import ListBank from './index';
import renderer from 'react-test-renderer';
import {store} from '@redux/store';

describe('<ListBank />', () => {
  it('<ListBank /> snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ListBank />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
