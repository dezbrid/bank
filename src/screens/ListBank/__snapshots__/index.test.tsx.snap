// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListBank /> <ListBank /> snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
